### PR TITLE
user signup flow completed with email verififcation

### PR DESCRIPTION
**Pull request by SunmadeBabafemi**
user signup flow completed.. to be tested with the lamda notification service

### CODE REVIEW

File: src/email-verification/email-verification.service.ts [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    const url = `${this.configService.get(
+      configConstant.baseUrls.identityService
+    )}/email-verification/${token}`;
+    const text = `Welcome to Zummit. To confirm your mail, please click the link below:\n\n\n ${url}`;
+
+    // An axios request to the notification service
+    const notification_url =  `${this.configService.get<string>(
+        configConstant.baseUrls.notificationService
+      )}/email/send-mail`
+      console.log(notification_url);
```
Comment (Chimexx) [inline src/email-verification/email-verification.service.ts:44]: pop

File: src/email-verification/email-verification.service.ts [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+    const sendNotification = await this.httpService.post(
+      notification_url,
+      {
+        email: email,
+        subject: 'Confirm Email',
+        text: text,
+      },
+    );
+    return await lastValueFrom(sendNotification.pipe());
```
Comment (Chimexx) [inline src/email-verification/email-verification.service.ts:55]: Any need of returning this? Just send

**Review by Chimexx — APPROVED** [review]
Works

File: src/auth/auth.service.ts [hunk 3]
```diff
@@ -4,24 +4,28 @@ import { InjectRepository } from '@nestjs/typeorm';
...
   constructor(
     @InjectRepository(User)
     private userRepo: Repository<User>,
+    private emailVerificationService: EmailVerificationService
     
   ){}
   async signup(user: UserSignupDto) {
     const userdata = Object.assign(new User(), user);
     const newUser = await this.userRepo.save(userdata).catch(async (e) => {
+      await this.emailVerificationService.resendVerificationLink(user.email);
```
[the remark this answers is not in the document]
Comment (SunmadeBabafemi) [reply src/auth/auth.service.ts:20]: hmmm.. i will look into it.. but i don't want to resend the link more than once... thanks for the observation

**Review by PhilemonBrain — APPROVED** [review]

**Pull request merged**